### PR TITLE
READY TO MERGE: Yaml driven config, redux

### PR DESCRIPTION
**Pull request by cmungall**
Supersedes #126

### CODE REVIEW

**Comment by cmungall** [comment]
This is ready to merge as soon as I figure out how to resolve. Some may involve manual porting of changes from a deleted file to the corresponding .jinja2 file.

This commit adds a README-developers and updates Changes.md, so this is a good place to start to understand the changes: https://github.com/INCATools/ontology-development-kit/pull/138/commits/c6d2a7bdda080005e49866c4b9210423b6cf78c1

File: template/.travis.yml.jinja2 [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 services:
   - docker
 
 before_install:
-  - docker pull obolibrary/osklite
+  - docker pull obolibrary/odkfull
 
 # command to run tests
-script: cd src/ontology && ./run.sh make test
+script: cd src/ontology && make test
```
Comment (matentzn) [inline template/.travis.yml.jinja2]: https://github.com/INCATools/ontology-development-kit/issues/111#issuecomment-426792147

**Comment by cmungall** [comment]
I agree with merging fast - would you be able to resolve the conflicts (I think other than the gitignore it should all be clobbers)

**Comment by matentzn** [comment]
Alright, done. Hope that didnt break anything, but a few local tests I ran seem to have worked.

**Pull request merged**